### PR TITLE
Fixed the intermittent test in #3035.

### PR DESCRIPTION
**Pull request by pelson**
The problem was that iris.util.format_array was stateful (via np.set_printoptions).

Follow up to: #3019
Closes: #3035

### CODE REVIEW

File: lib/iris/util.py [hunk 1]
```diff
@@ -948,16 +949,34 @@ def format_array(arr):
...
+
+    """
+    original_opts = np.get_printoptions()
+    np.set_printoptions(**kwargs)
+    try:
+        yield
+    finally:
+        np.set_printoptions(**original_opts)
+
+
```
Comment (bjlittle) [inline lib/iris/util.py]: @pelson Lovely. Great spot!

**Comment by pelson** [comment]
Oh dear... this has had a bit of a knock on. @bjlittle is going to take the baton on...

**Comment by bjlittle** [comment]
@pelson Okay, a bunch of changes that I banked due to this stateful behaviour... not to worry.

I'll fix them on your branch, since I've recently been in this space.

**Comment by pelson** [comment]
Hopefully this will now pass 🤞

**Comment by bjlittle** [comment]
You may need to rebase given the flurry of connected PRs that have just been banked.

We'll see what travis says...

**Comment by bjlittle** [comment]
:+1: Lovely, thanks @pelson

**Pull request merged**